data/yam/agama/auto/template.libsonnet

```diff
@@ -1,14 +1,19 @@
 local base_lib = import 'lib/base.libsonnet';
+local addons_lib = import 'lib/addons.libsonnet';
 local scripts_post_lib = import 'lib/scripts_post.libsonnet';
 local scripts_pre_lib = import 'lib/scripts_pre.libsonnet';
 local storage_lib = import 'lib/storage.libsonnet';
 
-function(bootloader=false, user=true, root=true, storage='', product='', registration_code='', scripts_pre='',
-  scripts_post='', encrypted=false) {
+function(bootloader=false, user=true, root=true, storage='', patterns='', product='', registration_code='',
+  registration_code_ha='', addons='', scripts_pre='', scripts_post='', encrypted=false) {
   [if bootloader == true then 'bootloader']: base_lib['bootloader'],
+  [if patterns != '' then 'software']: {
+    patterns: [patterns]
+  },
   [if product != '' then 'product']: {
     id: product,
     [if registration_code != '' then 'registrationCode']: registration_code,
+    [if addons == 'ha' then 'addons']: [ addons_lib.addons_ha(registration_code_ha) ],
```

Comment:
@okynos how we could pass the registration but at the same time having a comma separated list of addons here? because `addons` is only once in the jsonnet and then different combination of addons might be possible, wdyt?

Reply:
As my understanding is the 'addons' should be multiple for several addons added just like the autoyast.  If only one 'addons', how to separate different addons for id and registration password?

Reply:
Hello, I see in the code that you cannot add multiple addons including each registration code, I suggest an implementation like that https://github.com/os-autoinst/os-autoinst-distri-opensuse/compare/master...okynos:os-autoinst-distri-opensuse:addons-sles-ha?expand=1 take a look and let me know if it works for you

Reply:
Much appreciated the effort, @okynos could you please create an additional ticket to improve this, for now we only have one addon, I see that your solution is still quite complex, we need to keep information that is coupled in different array list, iterate with index (what can go wrong?), and use value of keys like registration code like they are, but other like `sles-ha` converted to `sles_ha`, we need something more obvious kind of something that add in the template:
if ha true, add this block with ha regcode, if we true, add this block with we recode,
but not sure at this point, better keep it as a future task.

Reply:
I updated PR with this, and VR passed https://openqa.suse.de/tests/17168131#step/boot_agama/1.